source/extensions/access_loggers/common/grpc_access_logger.h

```diff
@@ -135,10 +136,40 @@ template <typename LogRequest, typename LogResponse> class GrpcAccessLogClient {
     return true;
   }
 
+  Http::AsyncClient::StreamOptions createStreamOptionsForRetry() {
+    auto opt = Http::AsyncClient::StreamOptions();
+
+    if (!grpc_stream_retry_policy_) {
+      return opt;
+    }
+
+    envoy::config::route::v3::RetryPolicy retry_policy;
+    retry_policy.mutable_num_retries()->set_value(
+        PROTOBUF_GET_WRAPPED_OR_DEFAULT(*grpc_stream_retry_policy_, num_retries, 1));
+
+    if (grpc_stream_retry_policy_->has_retry_back_off()) {
+      const auto& base_retry_backoff = grpc_stream_retry_policy_->retry_back_off();
+      const auto base_interval_ms = PROTOBUF_GET_MS_REQUIRED(base_retry_backoff, base_interval);
+
+      auto* mutable_retry_back_off = retry_policy.mutable_retry_back_off();
+      mutable_retry_back_off->mutable_base_interval()->CopyFrom(
+          Protobuf::util::TimeUtil::MillisecondsToDuration(base_interval_ms));
+      mutable_retry_back_off->mutable_max_interval()->CopyFrom(
+          Protobuf::util::TimeUtil::MillisecondsToDuration(
+              PROTOBUF_GET_MS_OR_DEFAULT(base_retry_backoff, max_interval, 10 * base_interval_ms)));
+    }
+    retry_policy.set_retry_on("connect-failure");
```

Comment:
Why we need this part but not use `grpc_stream_retry_policy_` directly?

Reply:
this is because we'd like to set `RetryBackOff` but provided type is `BackoffStrategy`.

Reply:
add TODO comment to refactor here.